dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java

```diff
@@ -0,0 +1,583 @@
+	                    .showLinks(!excludeLinks)
+	                    .showExtensions(extensions)
+	                    .withFilter(filter)
+	                    .withHostOrFolderId(folderId)
+	                    .withLanguageId(languageId)
+	                    .offset(offset)
+	                    .showFiles(true)
+	                    .showPages(showPages)
+	                    .showFolders(!noFolders)
+	                    .showArchived(showArchived)
+	                    .showWorking(showWorking)
+	                    .sortBy(sortBy)
+	                    .sortByDesc(sortByDesc)
+	                    .withUser(user).build();
+	    
+	    return getFolderContent(browserQuery);
+
+	}
+	
+    @CloseDBIfOpened
+    public Map<String, Object> getFolderContent(final BrowserQuery browserQuery) throws DotSecurityException, DotDataException {
+
+
+        List<Map<String, Object>> returnList = new ArrayList<>();
+
+        Role[] roles = APILocator.getRoleAPI().loadRolesForUser(browserQuery.user.getUserId())
+                        .toArray(new Role[0]);
+
+
+        // gets folder parent
+        Folder parent = folderAPI.find(browserQuery.hostFolderId, browserQuery.user, false);
+
+
+        Host host = null;
+        if (parent == null) {// If we didn't find a parent folder lets verify if
+                             // this is a host
+            host = APILocator.getHostAPI().find(browserQuery.hostFolderId, browserQuery.user, false);
+
+            if (host == null) {
+                Logger.error(this, "Folder ID doesn't belong to a Folder nor a Host, id: " + browserQuery.hostFolderId
+                                + ", maybe the Folder was modified in the background.");
+                throw new NotFoundInDbException("Folder ID doesn't belong to a Folder nor a Host, id: " + browserQuery.hostFolderId);
+            }
+        }
+
+        if (browserQuery.showFolders) {
+            if (parent != null) {
+                List<Folder> folders = new ArrayList<Folder>();
+                try {
+                    folders = folderAPI.findSubFolders(parent, userAPI.getSystemUser(), false);
+                } catch (Exception e1) {
+                    Logger.error(this, "Could not load folders : ", e1);
+                }
+                for (Folder folder : folders) {
+                    List<Integer> permissions = permissionAPI.getPermissionIdsFromRoles(folder, roles, browserQuery.user);
+                    if (permissions.contains(PERMISSION_READ)) {
+                        Map<String, Object> folderMap = folder.getMap();
+                        folderMap.put("permissions", permissions);
+                        folderMap.put("parent", folder.getInode());
+                        folderMap.put("mimeType", "");
+                        folderMap.put("name", folder.getName());
+                        folderMap.put("title", folder.getName());
+                        folderMap.put("description", folder.getTitle());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "description" appears 5 times in this file; the first occurrence is on line 260](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)